### PR TITLE
Validate string length to repo field on create_request endpoint

### PR DESCRIPTION
**Pull request by fepas**
CLOUDBLD-7050

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] OpenAPI schema is updated (if applicable)
- [x] DB schema change has corresponding DB migration (if applicable)
- [x] README updated (if worker configuration changed, or if applicable)

### CODE REVIEW

**Comment by lkolacek** [comment]
@fepas thank you! PR looks good to me.

**Review by taylormadore — APPROVED** [review]
LGTM. Great job!

File: cachito/web/models.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ def is_request_ref_valid(ref: str) -> bool:
     return re.match(r"^[a-f0-9]{40}$", ref) is not None
 
 
+def is_request_repo_valid(repo: str) -> bool:
```
Comment (brunoapimentel) [inline cachito/web/models.py:29]: I was about to say that this function should be private... but looking at `is_request_ref_valid`, I saw it is used also to validate the `GET /api/v1/requests` endpoint.

So I'm thinking we should also use this function to validate the `repo` parameter in that endpoint.

Comment (fepas) [reply cachito/web/models.py:29]: I added the validation function in `/api/v1/requests` and changed the error message that existed in `ref`parameter validation (to keep the same message in all places). 

Also fixed the tests to work with the new error message and added a new test to check the `repo` parameter in this context.

**Review by brunoapimentel — APPROVED** [review]
LGTM :rocket:

**Pull request merged**